File: Second Year (2013-2014)/OMGL7/TP3 - Website in JSP/src/db/TitreDAO.java
package db;

import java.sql.*;
/**
* @author Andriamilanto Tompoariniaina - Canno Camille - Info2B
*/

/**
* Classe pour les objets DAO de Titre
*/
public class TitreDAO {
//Methodes
	/**
	*Constructeur
	*/
	public TitreDAO() {}

	/**
	*Pour retrouver tous les tuples
	*/
	public ResultSet findAll() {
		Statement stat = null;
		String query = "";
		ResultSet ret = null;

		try {
			//Recuperation de la connexion
			Connection con = MySQLConnection.getInstance().getConnection();

			//Preparation de la requete en ligne
			stat = con.createStatement();

			//Preparation de la requete
			query = "SELECT * FROM Titre;";

			//Le resultat a retourner
			ret = stat.executeQuery(query);
		}
		catch (SQLException e) {
			System.out.println("ERREUR: " + e.getMessage());
		}

		//Retourne l'execution de la requete sous la forme d'un objet ResultSet
		return ret;
	}

	/**
	*Permet de retrouver juste un tuple
	*@param id idTitre du titre a retrouver
	*/
	public Titre find(int id) {
		Statement stat = null;
		String query = "";
		Titre ret = new Titre();

		try {
			//Recuperation de la connexion
			Connection con = MySQLConnection.getInstance().getConnection();

			//Preparation de la requete en ligne
			stat = con.createStatement();

			//Preparation de la requete
			query = "SELECT * FROM Titre WHERE idTitre = " + id + ";";

			//Retourne l'execution de la requete sous la forme d'un objet ResultSet
			ResultSet result = stat.executeQuery(query);

			//Si le resultat est bon, prends la premiere ligne
			if (result.first()) {
				ret.init(id, result.getString(2), result.getFloat(3), result.getString(4), result.getInt(5), result.getInt(6));
			}
		}
		catch(SQLException e) {
			System.out.println("ERREUR: " + e.getMessage());
		}

		return ret;
	}


	/**
    *Methode qui permet d'inserer un tuple
    *@param tuple Objet de type Titre a inserer
    */
    public void insert(Titre tuple) {
    	Statement stat = null;
        String query = "";

        //Recuperation des attributs de l'objet Titre
        int idT = tuple.getIdTitre();
        String lib = tuple.getLibelle();
        float prix = tuple.getPrix();
        String artiste = tuple.getArtiste();
        int titreAlbum = tuple.getTitreAlbum();
        int titreCategorie = tuple.getTitreCategorie();

        try {
	    	//Recuperation de la connexion
	        Connection con = MySQLConnection.getInstance().getConnection();

	        //Preparation de la requete
	        query = "INSERT INTO Titre VALUES(" + idT + "," + lib + "," + prix + "," + artiste + "," + titreAlbum + "," + titreCategorie + ");";

	        //Execute la requête
			stat.executeQuery(query);
		}
		catch(SQLException e) {
			System.out.println("ERREUR: " + e.getMessage());
		}
    }


    /**
    * Permet de faire la mise a jour d'un tuple
    *@param tuple Objet nouveau a mettre a la place de l'ancien
    */
    public void update(Titre tuple) {
    	Statement stat = null;
        String query = "";

        //Recuperation des attributs de l'objet Titre
        int idT = tuple.getIdTitre();
        String lib = tuple.getLibelle();
        float prix = tuple.getPrix();
        String artiste = tuple.getArtiste();
        int titreAlbum = tuple.getTitreAlbum();
        int titreCategorie = tuple.getTitreCategorie();


        try {
	    	//Recuperation de la connexion
	        Connection con = MySQLConnection.getInstance().getConnection();

	        //Preparation de la requete
	        query = "UPDATE Titre SET Libelle = " + lib + ", Prix = " + prix + ", Artiste = " + artiste + ", TitreAlbum = " + titreAlbum + ", TitreCategorie = " + titreCategorie + " WHERE IdTitre = " + idT + ";";

	        //Execute la requête
			stat.executeQuery(query);
		}
		catch(SQLException e) {
			System.out.println("ERREUR: " + e.getMessage());
		}
    }


    /**
    * Permet de supprimer un tuple
    *@param idTitre Id du tuple a supprimer
    */
    public void delete(int idTitre) {
    	Statement stat = null;
        String query = "";

        try {
	    	//Recuperation de la connexion
	        Connection con = MySQLConnection.getInstance().getConnection();

	        //Preparation de la requete
	        query = "DELETE FROM Titre WHERE IdTitre = " + idTitre + ";";

	        //Execute la requête
			stat.executeQuery(query);
		}
		catch(SQLException e) {
			System.out.println("ERREUR: " + e.getMessage());
		}
    }
}

File: Second Year (2013-2014)/OMGL7/TP2 - Website in PHP/musiconline/musiconline/db/TitreDao.php
<?php
/* Authors: ANDRIAMILANTO Tompoariniaina && CANNO Camille
* Groupe: B1
* File: TitreDao.php
*/

//Les importations nécessaires
import ('phpens.framework.Application');
import ('phpens.framework.db.DataObject');
import ('phpens.framework.db.DataAccessObject');
import ('musiconline.db.Titre');


class TitreDao implements DataAccessObject{
  //Attribut instance de DataAccessObject
  private static $dao;


  //Constructeur vide
  private function __construct() { }

  //Retourne l'instance de TitreDAO
  public final static function getInstance() {

	//Vérifie si l'instance est correcte ou non
    if(!isset(self::$dao)) {

	  //S'il n'en est pas, il en créer un autre sans paramètres
      self::$dao= new TitreDao();
    }

    //S'il en est un, il le retourne, sinon il retourne donc une instance vide
    return self::$dao;
  }


  //Retourne toute la table
  public final function findAll(){

	 //Récupère la méthode de connection à la Base de donnée
     $dbc = Application::getInstance()->getDBConnection();

     //Variable où la requête est enregistrée
     $query = "select * from Titre";

     //Requête en dur, prends la bdd et y exécute la requête
     $stmt = $dbc->query($query);

     /*Récupère tout les tuples de Titre, et les transpose dans des instances de classe Titre
     Et fini par enregistrer le résultat dans $results, FALSE est retourné s'ya erreur */
     $results = $stmt->fetchALL(PDO::FETCH_CLASS, 'Titre');

     //Retourne le résultat de $results
     return $results;
   }


  //Permet d'ajouter un tuple
  public final function insert(DataObject $do){

	//Vérifie si l'objet entré en paramètre est instance de Titre
    if($do instanceof Titre){

	  //Récupère la méthode de connection à la BDD
      $dbc = Application::getInstance()->getDBConnection();

      //Prépare la requête SQL
      $query = "insert into Titre(IdTitre,Libelle,Prix,Artiste,TitreAlbum,TitreCategorie) values (:IdTitre,:Libelle,:Prix,:Artiste,:TitreAlbum,:TitreCategorie)";

	  //Prépare la requête en dur avec la BDD et la requête correspondante
      $stmt = $dbc->prepare($query);

      //Remplace tous les paramètres de Titre par les valeurs de l'objet
      $stmt->bindValue(':IdTitre',$do->getIDTitre(),PDO::PARAM_STR);
      $stmt->bindValue(':Libelle',$do->getLIbelle(),PDO::PARAM_STR);
      $stmt->bindValue(':Prix',$do->getPRix(),PDO::PARAM_STR);
      $stmt->bindValue(':Artiste',$do->getARtiste(),PDO::PARAM_STR);
      $stmt->bindValue(':TitreAlbum',$do->getTItreAlbum(),PDO::PARAM_STR);
      $stmt->bindValue(':TitreCategorie',$do->getTItreCategorie(),PDO::PARAM_STR);

     //Exécute finallement la requêtet
     $stmt->execute($data);
   }
 }


 //Permet de mettre à jour un tuple
  public function update(DataObject $obj){

	//Vérifie si l'objet entré en paramètre est instance de Titre
    if($obj instanceof Titre){

	  //Récupère la méthode de connection à la BDD
      $dbc = Application::getInstance()->getDBConnection();

      //Prépare la requête SQL
      $query = "UPDATE Titre SET Libelle = :Libelle, Prix = :Prix, Artiste = :Artiste, TitreAlbum = :TitreAlbum, TitreCategorie = :TitreCategorie WHERE IdTitre = :IdTitre";

	  //Prépare la requête en dur avec la BDD et la requête correspondante
      $stmt = $dbc->prepare($query);

      //Remplace tous les paramètres de Titre par les valeurs de l'objet
      $stmt->bindValue(':IdTitre',$obj->getIDTitre(),PDO::PARAM_STR);
      $stmt->bindValue(':Libelle',$obj->getLIbelle(),PDO::PARAM_STR);
      $stmt->bindValue(':Prix',$obj->getPRix(),PDO::PARAM_STR);
      $stmt->bindValue(':Artiste',$obj->getARtiste(),PDO::PARAM_STR);
      $stmt->bindValue(':TitreAlbum',$obj->getTItreAlbum(),PDO::PARAM_STR);
      $stmt->bindValue(':TitreCategorie',$obj->getTItreCategorie(),PDO::PARAM_STR);

     //Exécute finallement la requêtet
     $stmt->execute($data);
   }
  }


  //Permet de supprimer un tuple
  public function delete(DataObject $obj){

	   //Vérifie si l'objet entré en paramètre est instance de Titre
	   if($obj instanceof Titre){

	   //Récupère la méthode de connection à la BDD
      $dbc = Application::getInstance()->getDBConnection();

      //Prépare la requête SQL
      $query = "DELETE FROM Titre WHERE IdTitre = :IdTitre";

      //Prépare la requête en dur avec la BDD et la requête correspondante
      $stmt = $dbc->prepare($query);

      //Remplace les paramètres ":IdTitre" par la valeur de l'objet
      $stmt->bindValue(':IdTitre',$obj->getIDTitre(),PDO::PARAM_STR);

	 //Exécute finallement la requête
     $stmt->execute($data);
   }
  }
}
?>
